src/FunctionConfig.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { localize } from "./localize";
+
+interface IFunctionJson {
+    disabled?: boolean;
+    bindings?: IFunctionBinding[];
+}
+
+interface IFunctionBinding {
+    // tslint:disable-next-line:no-reserved-keywords
+    type?: string;
+    name?: string;
+    direction?: string;
+    authLevel?: string;
+    [propertyName: string]: string | undefined;
+}
+
+export enum HttpAuthLevel {
+    admin = 'admin',
+    function = 'function',
+    anonymous = 'anonymous'
+}
+
+enum BindingDirection {
+    in = 'in',
+    out = 'out'
+}
+
+/**
+ * Basic config for a function, stored in the 'function.json' file at the root of the function's folder
+ * Since the user can manually edit their 'function.json' file, we can't assume it will have the proper schema
+ */
+export class FunctionConfig {
+    public readonly functionJson: IFunctionJson;
+    public readonly disabled: boolean;
+    public readonly inBinding: IFunctionBinding;
+    public readonly inBindingType: string;
+    public readonly isHttpTrigger: boolean = false;
+    public readonly authLevel: HttpAuthLevel = HttpAuthLevel.function;
+
+    // tslint:disable-next-line:no-any
+    public constructor(data: any) {
+        let parseError: string | undefined;
+
+        this.functionJson = <IFunctionJson>data;
+        this.disabled = this.functionJson.disabled === true;
+
+        if (!this.functionJson.bindings || !(this.functionJson.bindings instanceof Array)) {
```

Comment:
I think it would be better if you verified data before assigning it to this.functionJson, so that after it's assigned we know it's valid.

In which case you should change IFunctionJson.bindings to be non-optional.